cmd/vicadmin/server.go

```diff
@@ -46,10 +55,78 @@ const (
 	formatZip
 )
 
+const sessionExpiration = time.Hour * 24
+const sessionCookieKey = "sessiondata"
+
+// UserSession holds a user's session metadata
+type UserSession struct {
+	username string
+	created  time.Time
+	config   *session.Config
+}
+
+// UserSessionStore holds and manages user sessions
+type UserSessionStore struct {
+	*sync.Mutex
```

Comment:
Why a pointer? Also if this object cannot be locked by callers, then you should make the lock an unexported field.